Ashbringer event sniffed

fangshun2004 opened the pull request:
<!-- First of all, THANK YOU for your contribution. -->
The generated coordinates of the Ashbringer event The sequence of actions uses the sniffing provided by @heyitsbench

## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address.

## Issues Addressed:
<!-- If your fix has a relating issue, link it below -->
- Closes 

## SOURCE:
<!-- If you can, include a source that can strengthen your claim -->
The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [x] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:
<!-- Does it build without errors? Did you test in-game? What did you test? On which OS did you test? Describe any other tests performed -->
This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.


## How to Test the Changes:
<!-- Describe in a detailed step-by-step order how to test the changes -->

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:
<!-- Is there anything else left to do after this PR? -->

- [ ] HIGHLORD_MOGRAINE's spell 28697 is too short for the spell visual effect when it hits MOGRAINE, and the visual effect disappears immediately after death, and other spell visual effects that also remain after death are required
- [ ] Fairbanks sniffed that the purification spell 28443 lacks visual effects and now uses 57767 as a visual effect
- [ ]

<!-- If you intend to contribute repeatedly to our project, it is a good idea to join our discord channel. We set ranks for our contributors and give them access to special resources or knowledge: https://discord.com/invite/DasJqPba)
     Do not remove the instructions below about testing, they will help users to test your PR -->
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

michaeldelago committed:
fix(Tools/DBImport): Add Env var support to DBimport tool (#17019)

add env var support for dbimport

Kitzunu committed:
refactor(docs): Clean up readme a bit (#16849)


---------

Co-authored-by: Benjamin Jackson <38561765+heyitsbench@users.noreply.github.com>

michaeldelago committed:
chore(apps): remove deprecated db_assembler and erroneous .gitmodules (#17039)

* chore(Contrib): remove deprecated db_assembler

* chore(repo): Remove erroneous gitmodule

grimgravy committed:
fix(Scripts/Quest): Improved Varedis Must Be Stopped (#16799)

* fix(Scripts/Quest): Varedis Must Be Stopped

* Update spell_quest.cpp

* Update data/sql/updates/pending_db_world/rev_1689813586760998100.sql

---------

Co-authored-by: Kitzunu <24550914+Kitzunu@users.noreply.github.com>

pvp100com committed:
fix(DB/Creature): Njorndar Proto-Drake vehicle unable to fly in “Vile Like Fire” (#16672)

Commit:
chore(DB): import pending files

Referenced commit(s): 6c049fcaa3a55f595f5cd006efa644659a752c0b

walkline committed:
fix(Core/Pets): Fix crash when pet tries to attack deleted object (#17034)

This situation occurs when, during a fight, a pet receives a command to apply a positive spell to an ally and then resume attacking its previous target.

walkline committed:
fix(Scripts/Dragonblight): Fix crash from Wintergarde Gryphon. (#17032)

fix(Vehicle): Fix crash on ExitVehicle from Wintergarde Gryphon.

Kitzunu committed:
fix(Core/Events): Implement Spirit of Competition event (#16963)

* fix(Core/Events): Implement Spirit of Competition event

Co-Authored-By: Benjamin Jackson <38561765+heyitsbench@users.noreply.github.com>

* Update Battleground.cpp

* Update rev_1691529514989936100.sql

* more stuff

* bye bye magic numbers

* fix dberrors

* Update Battleground.cpp

* Update Battleground.cpp

* Update Battleground.cpp

Co-Authored-By: Anton Popovichenko <walkline.ua@gmail.com>

* Update Battleground.cpp

* Update Battleground.cpp

* Update Battleground.cpp

* Update Battleground.cpp

* Update rev_1691529514989936100.sql

* commoneers

* Update src/server/game/Battlegrounds/Battleground.cpp

Co-authored-by: Anton Popovichenko <walkline.ua@gmail.com>

* ci

* Update rev_1691529514989936100.sql

* unnecessary whitespace

* Update src/server/game/Battlegrounds/Battleground.cpp

* Update src/server/game/Battlegrounds/Battleground.cpp

* Update src/server/game/Battlegrounds/Battleground.cpp

* Update Battleground.cpp

---------

Co-authored-by: Benjamin Jackson <38561765+heyitsbench@users.noreply.github.com>
Co-authored-by: Anton Popovichenko <walkline.ua@gmail.com>

Commit:
chore(DB): import pending files

Referenced commit(s): 6185b1d73b283a26a22d77c459ffd91a6cbd5457

michaeldelago committed:
feat(Apps/Docker): Use Env Vars for docker configuration (#17040)

* feat(docker): Use Env Vars for docker configuration

use env vars for docker

* simplify docker-compose.yaml

Nyeriah committed:
fix(Scripts/OHF): Don't update the escort steps when Drake dies (#17024)

* fix(Scripts/OHF): Don't update the escort steps when Drake dies

* update gossip req

* Update rev_1692233536025387200.sql

Commit:
chore(DB): import pending files

Referenced commit(s): 80a8f6d47a8a48869de197ed95c23b80c0999476

Tereneckla committed:
fix(Core/Pet): Correct (most) warlock pet scaling (#16769)

* create

* Update Pet.cpp

* remove unwanted change

* Update data/sql/updates/pending_db_world/rev_1689522771491547000.sql

* Update rev_1689522771491547000.sql

Commit:
chore(DB): import pending files

Referenced commit(s): cf87a2d8fea74ae0edef14e02e0365425bf45d29

pangolp committed:
fix(Script/Spells): Book of Fel Names (#17051)

Commit:
chore(DB): import pending files

Referenced commit(s): 957ccdb8453f9136cd2ae7c5ede453b5a126964a

Nyeriah committed:
fix(Scripts/Arcatraz): Fix not being able to try Skyriss a second time after wipes (#17018)

* fix(Scripts/Arcatraz): Revive Warden Mellichar on wipes at Skyriss

* Update boss_harbinger_skyriss.cpp

Kitzunu committed:
refactor(Misc/Conf): DBErrors is now Errors (#17053)

* refactor(Misc/Conf): DBErrors is now StartupErrors

* pt2

* Update ci-error-check.sh

* Update ci-error-check.sh

michaeldelago committed:
fix(Apps/Docker): ensure env var is used for db password (#17050)

Gultask committed:
feat(Core/SmartAI): Implement SMART_ACTION_FOLLOW_GROUP (#17033)

* Update SmartScript.cpp

* INIT

* add as coauthor as well :)

Co-Authored-By: cyberium <346907+cyberium@users.noreply.github.com>

* codestyle

* test?

---------

Co-authored-by: cyberium <346907+cyberium@users.noreply.github.com>

walkline committed:
fix(Core/Vehicles): Fix crash in Vehicle::AddPassenger (#17038)

* fix(Crash): Fix crash in Vehicle::AddPassenger

As I understand it, a try-catch block cannot intercept the ABORT function. Therefore, let's use a simple if statement instead.

* Address code review feedback

avarishd committed:
fix(DB/SAI): Correct Giselda the Crone's SAI. (#17008)

Commit:
chore(DB): import pending files

Referenced commit(s): 59db3b40ebb8b1cc6bf8b6c8b4cae32403c61bfe

Annamaria-CC committed:
fix(DB/Item): Deep Thunder ppm adjustment (#17049)

Commit:
chore(DB): import pending files

Referenced commit(s): 4544501b59b2a2c71d6a479e6d0177f76eade7c7

elthehablo committed:
fix(DB/SAI): Change Delmanis the Hated ReactState (#17059)

* initial

* initial

* typo

* change to simple update

* aggressive not defensive

Commit:
chore(DB): import pending files

Referenced commit(s): d77fc93f0a60f5e76e0fe3829d163fe7d1138cfa

elthehablo committed:
fix(DB/Creature): add `path_id` to some Ulduar mobs that were missing it (#17055)

* initial

* re-add auras

* REEEEEEEEEEEEEEEEE

Commit:
chore(DB): import pending files

Referenced commit(s): 8308e054b9edfbd7870989f500010084e9c377a2

elthehablo committed:
fix(DB/SAI): Adjust "Stinky"'s ReactState on waypoints (#17075)

* initial

* remove target storing and handle talking

so that all players in the group can complete the quest

* fix parameters

Commit:
chore(DB): import pending files

Referenced commit(s): 1702287967bae4b7d4ec2a644aaf82b64102bce9

elthehablo committed:
refactor(Scripts/SSC): modernise Morogrim Tidewalker with BossAI (#17028)

* fix ohf

* revert

* initial

* make bench happy

* make bench happier

The-GhostRider committed:
fix(Core/Item): Complimentary Brewfest Sampler (#16992)

* Fix(DB/C++):Item - Complimentary Brewfest Sampler

* Delete PlayerStorage.cpp

* Update rev_1691843684176706600.sql

Correction

* Ripristino il file PlayerStorage.cpp

* New file: SpellInfoCorrections.cpp

* Update rev_1691843684176706600.sql

Commit:
chore(DB): import pending files

Referenced commit(s): b1c8b0eaa26376ea5f4b54b640a967c6ec7b6fed

podmepodme committed:
fix(Scripts/Orgrimmar): replace hardcoded gossips in zone_orgrimmar (#16947)

* fix(CORE/Creature): replace hardcoded gossips in zone_orgrimmar

Closes AzerothCore issue #4179

* fix(CORE): Codestyle

* refactor(CORE/Character): Switch statement for thrall dialog

* fix(CORE): Codestyle

ChrisCGalbraith committed:
fix(Core/Spells): Add SPELL_ATTR7_DONT_CAUSE_SPELL_PUSHBACK to Hand of Sac and Divine Sac spells (#17041)

* add no spell pushback flag to auras

* Revert "add no spell pushback flag to auras"

This reverts commit 545e9fe4ee69bb079b69a4f2ba6039b834568a59.

* spellInfoCorrection for no pushback on HoS and Divine Sac

---------

Co-authored-by: ChrisCGalbraith <chriscgalbraith@gmai.com>

elthehablo committed:
fix(Scripts/Karazhan): allow sequence of releases to continue even if Dorothee dies early (#17077)

* initial
initial

Co-authored-by: danqu <3050620+danqu@users.noreply.github.com>

* some changes because reset dun work

* oops

* fixes hopefully

* debug

* more debug

* bruh

* cleanup

---------

Co-authored-by: danqu <3050620+danqu@users.noreply.github.com>

Synful-Syn committed:
fix(Core/Spell): Stop `Blink` from the falling player to move upward by 1y and stop moving backward when facing a wall (#16657)

* blink-bad-warps

* Change formatting of an if

walkline committed:
fix(Core/Scripts): Fix scripts where LoadObjectData can leave arrays boundaries. (#17070)

* fix(Core/Scripts): Fix scripts where LoadObjectData can leave arrays boundaries.

* Fix build

pvp100com committed:
fix(Core/MiscHandler): Dungeon requirements Message (#16701)

* fix(DB/Gameobject): fix towercanno error location in Flamewatch Tower

adjusted wintergrasp towercanno position

* fix(Core/MiscHandler): dungeon access requirements message parameter

requirements level variable always 0
requirements item cannot get from locale

Co-authored-by: aviar <aviar@sina.com>

* revert: random edits

---------

Co-authored-by: aviar <aviar@sina.com>
Co-authored-by: Stefano Borzì <stefanoborzi32@gmail.com>

Si1ker committed:
fix(DB/Creature): Add Taunt immunity to Magtheridon (#17076)

init

Commit:
chore(DB): import pending files

Referenced commit(s): cbbb705621f01f7cd216764daaaa3b1b5229caa6

elthehablo committed:
fix(DB/Creature): rework Agamand Family Crypt creature positions to make sure creatures don't aggro weirdly anymore (#17022)

* fix ohf

* revert

* initial

* the right sql file this TIME

help

* stop crying about codestyle

* frick YOU

* :(

* destroy readability - remove complaining? pog

* add new sql file with EOL converted

again... sigh

* Delete new-agamand-spawns.sql

* safety concerning primary key

delete based on guids instead of ids

Commit:
chore(DB): import pending files

Referenced commit(s): 122b7a8eb01e94d58b0a189479f2b2aeb4d7f0db

michaeldelago committed:
fix(Apps/Docker): ensure conf files always exist (#17071)

ChrisCGalbraith committed:
fix(Spell): set torment of the worgen to have 3% proc chance (#17005)


Co-authored-by: ChrisCGalbraith <chriscgalbraith@gmai.com>

Kitzunu committed:
chore(Core/Misc): Correct some fmt outputs (#17091)

elthehablo committed:
fix(Scripts/Karazhan): allow Moroes and adds to reset when an add is pulled out of the room (#17072)

* initial

* doforallsummons

* error fix

FrancescoBorzi committed:
refactor(Scripts): remove unused imports (#17097)

FrancescoBorzi committed:
refactor(Core): remove unused imports (#17094)

Kitzunu committed:
fix(Core/Crypto): fix crash (#17100)

Kitzunu committed:
chore(Deps/Misc): Print an EOL Deprecation warning for MySQL 5.7 and … (#17101)

chore(Deps/Misc): Print an EOL Deprecation warning for MySQL 5.7 and OpenSSL 1.1

The-GhostRider committed:
fix(DB/Conditions): quests 8353-8355-8356-8357-8311 (#17031)

Commit:
chore(DB): import pending files

Referenced commit(s): 228fe7274b2ffff00fea14f223ba23de04e38225

heyitsbench commented:
Remove every comment that includes text directly lifted from the parsed file.

Feel free to reword the comments to include a basic description, but do not include direct quotes from the file unless they are directly used in the cpp code.

elthehablo committed:
fix(Scripts/Karazhan): fix gossip from the Grandmother in the Big Bad Wolf fight (#17155)

* initial

* switch is better

* add correct menu ids cmangos cherrypick

...

Co-authored-by:  <7995382+killerwife@users.noreply.github.com>

* remove sql

Nyeriah committed:
fix(Scripts/Karazhan): Maiden of Virtue repentance should have a cast… (#17153)

fix(Scripts/Karazhan): Maiden of Virtue repentance should have a cast time

elthehablo committed:
fix(Scripts/Karazhan): make it so door closes when Big Bad Wolf is engaged in Little Red Riding Hood in the Opera (#17172)

* initial

* not a boolean

elthehablo committed:
fix(Scripts/Karazhan): change to right gossip menu for Barnes (#17151)

* initial

* make sense of ids

damnablebear committed:
fix(DB/Gameobject): Adjust position for unsniffed Truesilver node (#17112)

* fix(DB/Azshara): adjusting y and z axes for node

Adjusted position Y and position Z to make node spawn a bit more present and less above-ground than it was before. Sticks out of the ground an appropriate amount like other nodes now.

* Update truesilver-azshara-spawn-guid-9329-update.sql

* fix(DB/Azshara): adjusting y and z axes for node 

Moving file to the correct folder

* Delete truesilver-azshara-spawn-guid-9329-update.sql

had this file in the wrong directory

* changed eol to unix conversion

damnablebear committed:
fix(DB/Creature): Add CC immunities to Ethereum Prisoners (#17107)

* Add files via upload

fix(DB/Netherstorm): adding CC immunities to Ethereum Prisoners

* Update ethereum-prisoner-update.sql

* Update ethereum-prisoner-update.sql

* Update data/sql/updates/pending_db_world/ethereum-prisoner-update.sql

Co-authored-by: Dan <83884799+elthehablo@users.noreply.github.com>

---------

Co-authored-by: Dan <83884799+elthehablo@users.noreply.github.com>

Commit:
chore(DB): import pending files

Referenced commit(s): d514c52a9b3d58bba6e371589fd3059d8611fb87

elthehablo committed:
fix(DB/Creature): Add movement and events to Haggle in Deeprun Tram (#17095)

* initial

* update haggle

-fix query problems
-make wps more compact
-problem with not hitting certain events persists, I may have a solution coming

* fix timers so events fire correctly

* fix oops

Commit:
chore(DB): import pending files

Referenced commit(s): becf376b6915b86bede148ce0a0cb3ed81a1d81b

Nyeriah committed:
fix(Scripts/Karazhan): Fix Nightbane health phase percentages (#17173)

* fix(Scripts/Karazhan): Fix Nightbane health phase percentages

* Update boss_nightbane.cpp

Nyeriah committed:
fix(Scripts/Karazhan): Remove hacked knockback effect from Aran Flame… (#17174)

fix(Scripts/Karazhan): Remove hacked knockback effect from Aran Flame Wreath

Nyeriah committed:
fix(Scripts/Karazhan): Remove redundant Garrote cast and despawn gues… (#17152)

fix(Scripts/Karazhan): Remove redundant Garrote cast and despawn guests on evade

elthehablo committed:
fix(Scripts/MagtheridonsLair): make it so no attack is being done during Quake and fix p1 cheese tactic (#17136)

* initial

- adds condition for taunt speech
- makes it so no attack happens whatsoever during quake

* fix p1 channeler skip bug

* initialise channelers

* doforallminions

* thisiswhathappenswhenyoucodeinbrowser

* Update src/server/scripts/Outland/HellfireCitadel/MagtheridonsLair/boss_magtheridon.cpp

---------

Co-authored-by: Skjalf <47818697+Nyeriah@users.noreply.github.com>

Kitzunu committed:
fix(Core/Player): Allow to swap bags when they are filled (#17086)

walkline committed:
fix(Core/InstanceMap): Stop using deleted player in RemovePlayerFromMap (#17069)

Si1ker committed:
fix(Scripts/Magtheridon): Remove ThreatList before release (#17128)

init

elthehablo committed:
fix(Scripts/Ulduar): update Winter Jormungar and Snow Mound creature script (#17054)

* initial

* add auras

* spacing

Commit:
chore(DB): import pending files

Referenced commit(s): 189818e7813665f751119c090b55c3208ebba102

ChrisCGalbraith committed:
refactor(Scripts/TheEye): Refactor Void Reaver boss script to new scheduler (#17044)

* WiP draft of void reaver refactor

* linting

* remove UpdateAI override

* add delay after pounding

* separate arcane_orb into group to delay

* clang fix

* removed unused context param

---------

Co-authored-by: ChrisCGalbraith <chriscgalbraith@gmai.com>

elthehablo committed:
fix(Scripts/Magtheridon): add correct way to reset threat list (#17198)

* test

* fix but check

* fix?

Nyeriah committed:
feat(Core/Scripting): Implement OnAllowedToLootContainerCheck() hook (#17209)

* chore(Core/Loot): Move OnAllowedForPlayerLootCheck() hook call up

* Revert "chore(Core/Loot): Move OnAllowedForPlayerLootCheck() hook call up"

This reverts commit f841ef9da313976179fb788d9573ff8ced05ecb9.

* feat(Core/Scripting): Implement OnAllowedToLootContainerCheck() hook

* Update GlobalScript.cpp

Kitzunu committed:
fix(Core/AuctionHouse): fix crash in BuildListAuctionItems (#17222)

* closes https://github.com/azerothcore/azerothcore-wotlk/issues/17150

michaeldelago committed:
refactor(Apps/Dashboard): Remove Deno as a dependency to the dashboard (#16934)

* remove deno as a dependency

* use the ACORE_VERSION env var

* I forgot client-data

* accidentally removed too much

* docker-cmd: extraneous parallel flag

* typo

* remove docker image prune -af

the `docker image prune` removes the containers we just build.

* fetch everything in the beginning

byte-bandit committed:
fix(Core/Conf): log error in case configuration not found for DBImport (#17066)

* fix: log error in case configuration not found

* fix: register dbimport configuration file

michaeldelago committed:
fix(CI): pending sql changelog script's invocation (#17225)

fix deno run command

michaeldelago committed:
fix(CI): rebase instead of unshallow fetch (#17226)

Gultask committed:
feat(Core/SmartAI): Implement a few more stuff (#17090)

* init

* init2

* attempt to fix WP delays

* init3

* init4

* buildfix

* Create rev_1693093120417899200.sql

* workflow

* Underbat

* castflag power mana

* did not work

* Update SmartScriptMgr.cpp

* ugly

* port range to new system

* attempt

* simplify cast

simplify cast
add iteration for aura_not_present
recalc timer if already castig spell

* Update SmartScript.cpp

* attempt

* attempts

* HP PCT

* missing aura to targetselector

* Update rev_1693093120417899200.sql

* Update SmartScriptMgr.h

* Update SmartScriptMgr.h

* Update SmartScript.cpp

* remove not_attacking, fix behind_me

* revert is_behind_me

* Update SmartScriptMgr.h

* Update underbog.cpp

* Update underbog.cpp

Gultask committed:
feat(DB/Creature): Add CreateObject column in the creature table (#17187)

* Create rev_1693858384423226200.sql

* i forgor coauthor :skull:

Co-Authored-By: neifion-00000000 <91289495+neifion-00000000@users.noreply.github.com>

* Update rev_1693858384423226200.sql

* Update rev_1693858384423226200.sql

---------

Co-authored-by: neifion-00000000 <91289495+neifion-00000000@users.noreply.github.com>

Kitzunu committed:
fix(CI): pending files (#17229)

Commit:
chore(DB): import pending files

Referenced commit(s): 2698ed5a26c5aa786301d575b6bca19f17b97571

michaeldelago committed:
fix(apps/Docker): Missing arg for build:prod (#17231)

Kitzunu committed:
fix(CI): only run cpp check on AC (#17230)

heyitsbench committed:
chore(DB/Creature): Correct VerifiedBuild for unverified chess piece spells. (#17233)

Init.

Commit:
chore(DB): import pending files

Referenced commit(s): 6506a67d0f20e3a3db2ccd2dd0385eeb6151d58c

elthehablo committed:
fix(Scripts/Karazhan): Malchezaar infernals now spawn in the correct locations under the right conditions (#17237)

* initial
test

Co-authored-by: <100873791+Gultask@users.noreply.github.com>

* change spawnMask

* better failsafe

* get list once

* extend search range for relays

* put infernal spawn in separate method for

readability

* add more methods

* remove leftover method

* dumb and dumber mistakes

Commit:
chore(DB): import pending files

Referenced commit(s): a469cb5956e364058e5efc1869a8d40e4a5c5623

Nyeriah committed:
fix(Scripts/Karazhan): Should always be able to talk to chess pieces … (#17246)

fix(Scripts/Karazhan): Should always be able to talk to chess pieces regardless of faction

Commit:
chore(DB): import pending files

Referenced commit(s): 44d1df01c36d1a3c1f9e54c19b610b4f39c5a3d1

sudlud committed:
fix(Core/Achievement): Implement ACHIEVEMENT_CRITERIA_TYPE_OWN_RANK (#17180)

* fix(Core): Implement ACHIEVEMENT_CRITERIA_TYPE_OWN_RANK

* fix(Core): LoadAchievementCriteriaData(): Add ACHIEVEMENT_CRITERIA_TYPE_OWN_RANK

- fixes minor oversight of ACHIEVEMENT_CRITERIA_TYPE_OWN_RANK implementation
- this enables DB achievement_criteria_data check for ACHIEVEMENT_CRITERIA_TYPE_OWN_RANK

Commit:
chore(DB): import pending files

Referenced commit(s): 8ba8f0fe9e275dbc7c4a5f63b46d6749b532ec52

Kitzunu committed:
chore(Core/Misc): Silence compile warning in boost::process with vers… (#17251)

chore(Core/Misc): Silence compile warning in boost::process with version 1.83

* Cherry-pick commit https://github.com/TrinityCore/TrinityCore/commit/d8c98579d4a2c4191e42c2542901274ffe53083c

Co-authored-by: Shauren <shauren.trinity@gmail.com>

Kitzunu committed:
chore(CI): Update deps for Windows and Mac builds (#17252)

* chore(CI): Update deps for Windows and Mac builds

* Update windows.sh

* Update windows.sh

Kitzunu committed:
chore(docs): remove reference to BountySource (#17253)

Gultask committed:
feat(DB): Add another column for comments (#17241)

* Create rev_1694636334573006200.sql

* varchar -> text

* Update rev_1694636334573006200.sql

Commit:
chore(DB): import pending files

Referenced commit(s): 5d128d38ea98c325422d0c81e56d7743557216d0

heyitsbench committed:
refactor(DB/Player): Unify two player stat tables. (#17188)

* Init.

Change query table names, move data to existing table.

* Combine two player stat loading functions.

Commit:
chore(DB): import pending files

Referenced commit(s): 358a1e71e0b84c194164d0fb74fd2372d9bcf713

Nyeriah committed:
fix(DB/Creature): Correct Orc Warlock controlled spell (#17232)

* fix(DB/Creature): Correct Orc Warlock controlled spell

* Update data/sql/updates/pending_db_world/rev_1694389067923549800.sql

* Update data/sql/updates/pending_db_world/rev_1694389067923549800.sql

* Update data/sql/updates/pending_db_world/rev_1694389067923549800.sql

---------

Co-authored-by: Kitzunu <24550914+Kitzunu@users.noreply.github.com>

Commit:
chore(DB): import pending files

Referenced commit(s): f2de40d1721aa45da7d804ed2e45373d84329112

elthehablo committed:
fix(Scripts/Karazhan): decrease Aran cast cooldown on normal spells (#17260)

initial

Tereneckla committed:
refactor(Scripts/Spells): Check for existence of AD debuff instead of adding a custom cooldown to heal (#17265)

* Update spell_paladin.cpp

* whitespace

Kitzunu committed:
fix(Scripts/Chat): Fix chat logger (#17102)

* fix(Scripts/Chat): Fix chat logger

* Update chat_log.cpp

* Update worldserver.conf.dist

* Update chat_log.cpp

* fix build

* oopsie

* fix clang

Kitzunu committed:
chore(Core/enumutils): regenerate enumutils (#17087)

* chore(misc/enumutils): update enumutils

* fix build

* latest update

Kitzunu committed:
fix(DB/Character): Allow profanity_name and reserved_name to be accen… (#17224)

fix(DB/Character): Allow profanity_name and reserved_name to be accent sensitive

* closes https://github.com/azerothcore/azerothcore-wotlk/issues/13214

Commit:
chore(DB): import pending files

Referenced commit(s): 4f6a402da3d08310bc33e6ba0362b3a7a9347fb1

Kitzunu committed:
chore(Scripts/DK): Move CheckCast to spell_dk_raise_ally script (#17221)

* chore(Scripts/Spell): Move CheckCast to spell_dk_raise_ally script

* Update spell_dk.cpp

* Update spell_dk.cpp

neifion-00000000 committed:
fix(DB/Creature): Modernize and follow-up on PR #9912 #9943 #10052 #10282 #10285 #12702 #12751 #12765 #12854 (#17217)

* fix(db/creature): Modernize PR #9912

* Update rev_1694148121469432000.sql

* Update rev_1694148121469432000.sql

typo

* Update rev_1694148121469432000.sql

* Update rev_1694148121469432000.sql

---------

Co-authored-by: Kitzunu <24550914+Kitzunu@users.noreply.github.com>

Gultask committed:
fix(DB/Creature): Netherspite is immune to taunts (#17261)

Create rev_1694778808871595100.sql

Co-authored-by: Benjamin Jackson <38561765+heyitsbench@users.noreply.github.com>

Commit:
chore(DB): import pending files

Referenced commit(s): 63a4301b1c87dd7086fc6af2982d38c440c34fe2

elthehablo committed:
fix(Scripts/Karazhan): remove redundant spellcasting check on Netherspite to ensure he moves back to phase 1 (#17269)

initial

elthehablo committed:
fix(DB/Karazhan): fix Curator's patrol movement using sniffs (#17234)

initial

Nefertumm committed:
fix(Scripts/MagtheridonLair): Magtheridon Debris (#17249)

* fix(Core/Scripts): Magtheridon Debris

* unhack

* cleanup

* eof line

* unused variable

* more cleanup

Commit:
chore(DB): import pending files

Referenced commit(s): c149f213fe9f10a92b02640432f9f68453f9f9a1

elthehablo committed:
refactor(Scripts/SSC): Hydross the Unstable boss script updated (#17081)

* initial

* fix timers

Water tomb and vile sludge timers homogenised. I did this because the sniffs turn him into the pure form quite quickly. So I am assuming the initial timer is the same for both, and then taking some kind of average for the water tomb and also using that for vile sludge. We may need testing where we keep him in the first form for a bit just to look at the timers

Gultask committed:
fix(DB/Creature): The Big Bad Wolf should be Taunt immune (#17255)

Create rev_1694739263824086500.sql

Commit:
chore(DB): import pending files

Referenced commit(s): a22a04908a4902e59cc71eeac29024d32bd4666e

elthehablo committed:
fix(Scripts/BlackwingLair): Orb of Domination can no longer be used when you are controlling a pet (#16880)

initial

Gultask committed:
fix(DB/Creature): Ravenous Flayer Matriarch (#17256)

Create rev_1694740225097878900.sql

Commit:
chore(DB): import pending files

Referenced commit(s): a7d2f2effae62483c8ddcdd6038da23d024d371a

elthehablo committed:
fix(Scripts/Darkshore): ensure Kerlonian Evershade is always in bear form (#17062)

initial

Commit:
chore(DB): import pending files

Referenced commit(s): 45d6009cb7189706afe58812155b3d7bd60174dd

Kitzunu committed:
fix(Core/TaskScheduler): warning (#17275)

Nyeriah committed:
fix(Core/GameObject): Also send EventInform() calls for GAMEOBJECT_TY… (#17283)

fix(Core/GameObject): Also send EventInform() calls for GAMEOBJECT_TYPE_CAMERA

- This allows object and zone AI to process events called by camera objects as well (previously only possible through event_scripts)

(cherry-picked from commit https://github.com/TrinityCore/TrinityCore/commit/6904073c9cf2d6d1291bf976b8efff55edae74bc)

Kitzunu committed:
chore(misc): cleanup (#17274)

Kitzunu committed:
chore(Worldserver/Config): Clean up and reorganize config file (#17272)

* chore(Worldserver/Config): Clean up and reorganize config file

* Update worldserver.conf.dist

* \n\n\n

Gultask committed:
fix(DB/SAI): Rework Sethekk Halls SAI (#17257)

* Create rev_1694744446848459200.sql

* Update rev_1694744446848459200.sql

Co-Authored-By: offl <11556157+offl@users.noreply.github.com>

* Update rev_1694744446848459200.sql

Co-Authored-By: offl <11556157+offl@users.noreply.github.com>

* Update rev_1694744446848459200.sql

Co-Authored-By: offl <11556157+offl@users.noreply.github.com>

---------

Co-authored-by: offl <11556157+offl@users.noreply.github.com>

Tereneckla committed:
fix(Scripts/Karazhan): Curator fixes (#17270)

* stop channeling on enrage

* Create rev_1694861198470600600.sql

* Update boss_curator.cpp

* Update src/server/scripts/EasternKingdoms/Karazhan/boss_curator.cpp

Co-authored-by: Kitzunu <24550914+Kitzunu@users.noreply.github.com>

---------

Co-authored-by: Angelo Venturini <nefertum.dev@protonmail.com>
Co-authored-by: Kitzunu <24550914+Kitzunu@users.noreply.github.com>

Commit:
chore(DB): import pending files

Referenced commit(s): c709e2e7d3ba426178c2eebce30f6d30da71ff7c

Nyeriah committed:
feat(Core/Instance): Add TaskScheduler to the InstanceScript class (#17284)

feat(Core/Instance): Add TaskScheduler to the instancescript class

elthehablo committed:
fix(Scripts/Karazhan): Shade of Aran now no longer casts spells whilst he should be drinking (#17282)

* initial

* remove redundancy

* remove method

* space gone

* indentación

Kitzunu committed:
chore(misc): Mordenize maths (#17273)

FireCulex committed:
fix(Core/Item): Reduce amount of excessive logging for enforeDBCAttributes (#16508)

bypass code block if enforceDBCAttributes is set to false

grimgravy committed:
fix(Scrpits/Quest): Improve Recharging the Batteries (#16597)

* fix(Scrpits/Quest): Improve Recharging the Batteries

* Update zone_netherstorm.cpp

* update

ljfp committed:
fix(DB/Creature): Removed duplicated fire festival Flame Wardens (#16671)

* fix(DB/GameEventCreature): Removed duplicated fire festival Flame Guards.

* Also removed the NPCs from the creatures table

* Removed reference to old table.

Nyeriah committed:
fix(Scripts/Karazhan): Fix Enfeeble hitting more than 5 targets (#17290)

* fix(Scripts/Karazhan): Fix Enfeeble hitting more than 5 targets

* reorder so there are no cases where only 4 players are hit

* missing override

TuriansNotBad committed:
fix(Core/Position): Normalize Position.GetRelativeAngle (#16825)

Normalize GetRelativeAngle

Normalize angle returned by both Position.GetRelativeAngle functions.

Commit:
chore(DB): import pending files

Referenced commit(s): 24c97c5f2f932ac7f851a31500ebaa5f46b61b67

fangshun2004 committed:
fix(DB/Creature): move Fairbanks gossip to the database (#16974)

* Add gossip_menu_id for fairbanks

* npc_text ID uses the npc_text_broadcast_text Datasheet ID in CMANGOS

* Some emoticons will hold the weapon in your hand when used, so use smartAI instead.

* Only if there is an inventory or equipment Ashbringer will the gossip_menu_option be displayed

* Remove the code from the script

* Update the data

* Code formatting Remove comments

* Replace ## with --
Replace npc_text_broadcast_text with npc_text_broa*dcast_text

* Replace npc_text_broadcast_text with npc_text_broa*dcast_text

* Modify the comment

Commit:
chore(DB): import pending files

Referenced commit(s): ca95b4a12965c25f74712d9e9f50ea83bc0e2106

Kitzunu committed:
chore(CI): Hello Github Bot and Update actions version (#17285)

Tony931023 committed:
fix(DB/Quest) Know Thine Enemy RP event (#16258)

* Create 62896.sql

Co-Authored-By: Rushor <Rushor@users.noreply.github.com>

* Update 62896.sql

---------

Co-authored-by: Rushor <Rushor@users.noreply.github.com>
Co-authored-by: Kitzunu <24550914+Kitzunu@users.noreply.github.com>

Commit:
chore(DB): import pending files

Referenced commit(s): 9523081c3253b10140ce03832b0b727e8a976a5a

fangshun2004 committed:
fix(DB/Creature): Godrick Farsan event (#16607)

* Added emoticons and dialogue events to Warlock Quest NPC Godrick Farsan (ID 5693).

* Update UnitFlags

* Modify the delay

* Fix syntax errors

* Try the fix again

* The delay of task emoticons is increased by 200ms, preventing the loss of missing emoticons when preparing tasks after cache clearing

* Update unit_flags in smatscript

* Update the comment

* quest_offer_reward emote ported to SmartAI

* Remove Directional Coordinate Update and wait for sniffing
set_unit_flag Update to use SMATAI

* Update rev_1687513981074967100.sql

---------

Co-authored-by: Kitzunu <24550914+Kitzunu@users.noreply.github.com>

Nyeriah committed:
fix(Scripts/MagistersTerrace): Script Kalecgos' appearance after usin… (#17286)

Gultask committed:
fix(DB/SAI): Worldserver Error in Dalaran (#17292)


Co-authored-by: daobashun <49193927+fangshun2004@users.noreply.github.com>

fangshun2004 committed:
sniffed

fangshun2004 commented:
https://github.com/feltwitch/azerothcore-wotlk/commit/133e2b746770e74967035f52de08ae0adff48af9 Time stamp with sniffing

fangshun2004 committed:
1.Use farther sniffing coordinates
2.Fix Incorrect time to move to target location
3.Fixed "HIGHLORD_MOGRAINE" movement speed is the same as the player
3.When "HIGHLORD_MOGRAINE" is summoned Reset Mograine (kill and resurrect) will Despawn "HIGHLORD_MOGRAINE" (when debugging)
